### ISSUE TITLE
Configuration cache state could not be cached

### ISSUE THREAD

**Issue by rbro112** [EmergeTools/emerge-android#115]
Running a task with configuration cache enabled results in a violation:

Command:
```
./gradlew :app:emergeUploadSnapshotBundleGooglePlayDebug
```

Output:
```
Configuration cache state could not be cached: field `artifactMetadata$delegate` of task `:app:emergeUploadSnapshotBundleGooglePlayDebug` of type `com.emergetools.android.gradle.tasks.snapshots.UploadSnapshotBundle`: error writing value of type 'kotlin.SynchronizedLazyImpl'
> Expected file collection to contain exactly one file, however, it contains no files.
```

**Comment by ankushg** [comment]
It seems that Gradle is treating [`artifactMetadata`](https://github.com/EmergeTools/emerge-android/blob/8174b9c6e2d7791e01363aabd4d287736bfa38c4/gradle-plugin/plugin/src/main/kotlin/com/emergetools/android/gradle/tasks/snapshots/UploadSnapshotBundle.kt#L40) when it's serializing the task's state for the config cache, when it's really just a lazily computed value that's needed at execution-time

Maybe marking it with `@Transient` would help? https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.jvm/-transient/

**Comment by rbro112** [comment]
Closing as fixed with 3.0.0 release
